app/include/user_config.h

```diff
@@ -11,11 +11,13 @@
 // #define FLASH_16M
 #define FLASH_AUTOSIZE
 #define FLASH_SAFE_API
-// #define DEVELOP_VERSION
-#define FULL_VERSION_FOR_USER
 
-#define USE_OPTIMIZE_PRINTF
+// Byte 107 of esp_init_data_default, only one of these 3 can be picked
+#define ESP_INIT_DATA_ENABLE_READVDD33
+//#define ESP_INIT_DATA_ENABLE_READADC
```

Comment:
Hi
Thanks another helpful user pointed me in the right direction.

I loaded Oracle VM Virtual box Ubuntu all the Compilers and git hubbed 
the source,
Changed the setting in config.h and compiled ,FaBBed it and its all 
Working --   ( FaBBed ) Flag and Basic Burn)

Another user also pointed out the route of using the the EMU Flash tool 
, And that was very helpfully

I was asking from the point of the experienced users which is option 
they find better to use and has the least problems .

Thank you gave me all the options , I was hoping for an opinion on which 
was for you the better option.
It did help.

Thanks I should pay attention to what I am doing more often.
Missed that it was 2 months old .

I should read the manual more often as I was merely FaBBing and burning 
Lua_ing my ESP when i came across the 65535 read glitch when I had just 
read the VCC of my unit and needed an external Current read before I 
sent the data to my rest service and before sending the unit to sleep   .

So I blindly asked a question,

*I am mere Padawan*

Respectfully I bid you adieu

Regards JarrodMP


On 2016/02/06 21:35 PM, Marcel Stör wrote:
>
> In app/include/user_config.h 
> <https://github.com/nodemcu/nodemcu-firmware/pull/746#discussion_r52106356>:
>
> >
> > -#define USE_OPTIMIZE_PRINTF
> > +// Byte 107 of esp_init_data_default, only one of these 3 can be picked
> > +#define ESP_INIT_DATA_ENABLE_READVDD33
> > +//#define ESP_INIT_DATA_ENABLE_READADC
>
>     I would like to build the firmware...what or how is the best root
>     to take?
>
> Not to ask such questions in a 2-months old commit comment would be a 
> good start. Then you can familiarize yourself with 
> http://nodemcu.readthedocs.org/en/dev/en/build/.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/nodemcu/nodemcu-firmware/pull/746/files#r52106356>.
>
> No virus found in this message.
> Checked by AVG - www.avg.com <http://www.avg.com>
> Version: 2016.0.7357 / Virus Database: 4522/11566 - Release Date: 02/06/16
>